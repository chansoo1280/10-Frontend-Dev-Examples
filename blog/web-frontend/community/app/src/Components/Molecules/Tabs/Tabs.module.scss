@import "@Styles/scss/default.scss";

.tabs {
    gap: 32px;

    &__tab {
        padding-left: 0;
        padding-right: 0;

        &:focus,
        &--active,
        &--active:focus,
        &--active:hover {
            color: $primary-6;
            border-bottom: 2px solid $primary-6;
        }
    }
}