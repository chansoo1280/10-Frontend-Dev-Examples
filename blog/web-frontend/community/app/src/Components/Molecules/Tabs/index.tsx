// #region Global Imports
import React from "react"
import classNames from "classnames"
// #endregion Global Imports

// #region Local Imports
import { defaultProps, Space, Button } from "@Components"
import styles from "./Tabs.module.scss"
// #endregion Local Imports

export interface Tab {
    title: string
    href?: string
}
interface TabsProps extends defaultProps {
    activeIdx: number | null
    tabList: Tab[]
    onClick: (tab: Tab, idx: number) => void
}

const Tabs = (props: TabsProps): JSX.Element => {
    const { activeIdx, tabList, show, className, onClick, ...rest } = props
    const prefixCls = "tabs"
    const classes = classNames(
        styles[`${prefixCls}`],
        {
            [styles[`${prefixCls}--hide`]]: show === false,
        },
        className,
    )
    return (
        <Space className={classes} {...rest}>
            {tabList.map((tab, idx) => (
                <Button
                    size="large"
                    type="text"
                    onClick={() => {
                        onClick(tab, idx)
                    }}
                    key={tab.title}
                    href={tab.href}
                    className={classNames(styles[`${prefixCls}__tab`], {
                        [styles[`${prefixCls}__tab--active`]]: activeIdx === idx,
                    })}
                >
                    {tab.title}
                </Button>
            ))}
        </Space>
    )
}
export default Tabs
