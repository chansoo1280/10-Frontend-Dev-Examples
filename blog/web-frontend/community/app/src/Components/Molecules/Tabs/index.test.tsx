import { fireEvent, render, screen } from "@testing-library/react"
import Tabs from "./"
describe("Tabs", () => {
    it("renders tabs unchanged", () => {
        const handleClick = jest.fn()
        const { container } = render(<Tabs activeIdx={0} onClick={handleClick} tabList={[{ title: "Tab 1" }]} />)
        expect(container).toMatchSnapshot()
    })
    it("should render a tabs with the class", () => {
        const handleClick = jest.fn()
        render(<Tabs activeIdx={0} data-testid="tabs" onClick={handleClick} tabList={[{ title: "Tab 1" }]} />)

        const tabs = screen.getByTestId("tabs")
        expect(tabs).toBeInTheDocument()
        expect(tabs).toHaveClass("tabs")
    })
    it("should render a tabs, checks for onClick attribute", () => {
        const handleClick = jest.fn()
        render(<Tabs activeIdx={0} data-testid="tabs" onClick={handleClick} tabList={[{ title: "Tab 1" }]} />)

        const button = screen.getByText("Tab 1")
        fireEvent.click(button)
        expect(handleClick).toBeCalledTimes(1)
    })
})
