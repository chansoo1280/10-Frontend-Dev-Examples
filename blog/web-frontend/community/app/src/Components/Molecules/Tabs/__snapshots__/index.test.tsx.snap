// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders tabs unchanged 1`] = `
<div>
  <div
    class="space space--wide tabs"
    style="gap: 32px; padding: 0px 16px;"
  >
    <button
      class="btn btn--large btn--text tabs__tab tabs__tab--active"
      data-click-animating="false"
      type="button"
    >
      Tab 1
    </button>
  </div>
</div>
`;
